### ISSUE TITLE
Only the first checkpoint is being saved in model_dir

### ISSUE THREAD

**Issue by Pari-singh** [google/automl#738]
Hello, 

I am training my data with coco weights for efficientdet-d7x using train_and_eval, for some reason, the epoch checkpoints are not getting saved in model-dir, I only see the 0th checkpoint along with bunch of tfevents and one graph.pbtxt. I just have 
model.ckpt-0.data-00000-of-00001, model.ckpt-0.index, model.ckpt-0.meta 
as checkpoints even after 40 epoch training. Could someone tell what's going wrong?

**Comment by Pari-singh** [comment]
@mingxingtan

**Comment by mingxingtan** [comment]
What's your command line?

I just tried the latest code and it has multiple checkpoints.

**Comment by Pari-singh** [comment]
Hi, thanks for the reply. I remember reading the same issue somewhere else too, where the events is created but the checkpoint saved is only the 0th one.
I used this command to finetune on BDD dataset
export CUDA_VISIBLE_DEVICES=0; python main.py --mode=train --training_file_pattern=/media/data_1/BDD100K/tfrecord_new/100k/train/output_train100k__000*.tfrecord --validation_file_pattern=/media/data_1/BDD100K/tfrecord_new/100k/val/output_val100k__0000*.tfrecord --model_name=efficientdet-d7x --model_dir=my_trainings/BDD_finetune --ckpt=efficientdet-d7x --train_batch_size=16 --eval_batch_size=1 --eval_samples=10000 --num_examples_per_epoch=70000 --num_epochs=300 --hparams=bdd_config.yaml

Another issue I face is with train batch size 16, I am running into NaN loss error no matter how low I give the learning rate.
Currently my bdd_config.yaml looks like:
num_classes: 11
var_freeze_expr: '(efficientnet|fpn_cells|resample_p6)'
label_map: {1: person, 2: rider, 3: car, 4: bus, 5: truck, 6: bike, 7: motor, 8: traffic light, 9: traffic sign, 10: train}
mixed_precision: True
learning_rate: 0.00001
lr_warmup_init: 0.0001

Looking forward for support. Thanks

**Comment by Pari-singh** [comment]
The new changes seems to have resolved this issue. I will close this issue.
